metagraph/src/annotation/annotation_converters.cpp

```diff
@@ -983,134 +1080,339 @@ void convert_to_row_annotator(const ColumnCompressed<std::string> &source,
                               RowCompressed<std::string> *annotator,
                               size_t num_threads);
 
-std::unique_ptr<RowDiffAnnotator> convert_to_row_diff(const graph::DBGSuccinct &graph,
-                                                      RowCompressed<std::string> &&annotation,
-                                                      uint32_t num_threads,
-                                                      uint32_t max_depth) {
-    assert(graph.num_nodes() == annotation.num_objects());
-    uint64_t nnodes = graph.num_nodes();
-    Vector<uint64_t> node_diffs;
-    Vector<std::pair<uint64_t, uint32_t>> pos(nnodes, { 0, 0 });
-    sdsl::bit_vector terminal;
-
-    std::atomic<uint64_t> terminal_count = 0;
-    std::atomic<uint64_t> forced_terminal_count = 0;
-    std::atomic<uint64_t> depth_terminal_count = 0;
-    std::atomic<uint64_t> boundary_size = 0;
-    std::atomic<uint64_t> visited_nodes = 0;
-    graph.get_boss().call_sequences_row_diff(
-        [&](const std::vector<uint64_t> &path, std::optional<uint64_t> anchor) {
-            assert(!path.empty());
-            std::vector<uint64_t> anno_ids(path.size());
-            for (uint32_t i = 0; i < path.size(); ++i) {
-                uint64_t kmer_index = graph.boss_to_kmer_index(path[i]);
-                assert(kmer_index <= nnodes);
-                anno_ids[i] = graph::AnnotatedSequenceGraph::graph_to_anno_index(kmer_index);
+using CallOnes = std::function<void(const bit_vector &source_col,
+                                    uint64_t row_idx,
+                                    uint64_t start_idx,
+                                    uint64_t source_idx,
+                                    uint64_t col_idx,
+                                    const std::vector<uint64_t> &succ_chunk,
+                                    const std::vector<uint64_t> &pred_chunk,
+                                    const std::vector<uint64_t> &pred_chunk_idx)>;
+
+constexpr uint64_t chunk_size = 1 << 20;
+
+void traverse_anno_chunked(const std::string &name,
+                   const graph::DBGSuccinct &graph,
+                   const std::string &graph_fname,
```

Comment:
why do you need to pass both graph and its filename?

Reply:
Could also be `anchors_filename`, or something. I wanted to point out that this doesn't have to be in the same directory as the graph. They could be stored in different locations. So the name should reflect that. `outfbase` is fine too